### ISSUE TITLE
Container errors in controllers with validation rules.

### ISSUE THREAD

**Issue by brendt** [nunomaduro/larastan#82]
Almost all my controllers generate errors, saying they try to resolve an undefined class:

```
  Line   app/Http/Admin/Controllers/Settings/UnitAssignmentOrderController.php
 ------ -----------------------------------------------------------------------
         Internal error: Class unit_types does not exist
         Run PHPStan with --debug option and post the stack trace to:
         https://github.com/phpstan/phpstan/issues/new
```

```
  Line   app/Http/Admin/Controllers/Settings/UnitPricesController.php
 ------ --------------------------------------------------------------
         Internal error: Class message does not exist
         Run PHPStan with --debug option and post the stack trace to:
         https://github.com/phpstan/phpstan/issues/new
```

```
  Line   app/Http/Admin/Controllers/Settings/InvoiceLabelsController.php
 ------ -----------------------------------------------------------------
         Internal error: Class labels does not exist
         Run PHPStan with --debug option and post the stack trace to:
         https://github.com/phpstan/phpstan/issues/new
```

etc.

Running with `--debug` shows this:

```php
 ReflectionException  : Class labels does not exist

  at /vendor/laravel/framework/src/Illuminate/Container/Container.php: 767
  763:         if ($concrete instanceof Closure) {
  764:             return $concrete($this, $this->getLastParameterOverride());
  765:         }
  766:
  767:         $reflector = new ReflectionClass($concrete);
  768:
  769:         // If the type is not instantiable, the developer is attempting to resolve
  770:         // an abstract type such as an Interface of Abstract Class and there is
  771:         // no binding registered for the abstractions so we need to bail out.
  772:         if (! $reflector->isInstantiable()) {

  Exception trace:

  1   ReflectionClass::__construct("labels")
      /vendor/laravel/framework/src/Illuminate/Container/Container.php : 767

  2   Illuminate\Container\Container::build("labels")
      /vendor/laravel/framework/src/Illuminate/Container/Container.php : 646
```

I'm not 100% sure where this error comes from. It seems to have something to do with request validation:

```php
$validated = collect($request->validate($this->rules()));

// ...

protected function rules(): array
{
    return [
        'labels' => 'array',
        // ...
    ];
}
```

All these fields seem to be the first entry in the validation rules, probably it has something to do with the `->validate` macro on `Illuminate\Http\Request` ?

**Comment by canvural** [comment]
I guess this is the same as #77

**Comment by szepeviktor** [comment]
> I guess this is the same as #77

Yes, array elements/keys are recognized as classes.

**Comment by szepeviktor** [comment]
Solved in https://github.com/nunomaduro/larastan/commit/84cd60fd17f191a631580ba20f2e76b76c0b3cb7

**Comment by nunomaduro** [comment]
@brendt @canvural @szepeviktor Thanks for reporting this issue.

Is now fixed on the version `v0.2.2`. Do you mind of performing a quick test?

**Comment by brendt** [comment]
It works! 👌
